x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlExecutionInfo.java

```diff
@@ -174,6 +174,9 @@ public TimeValue planningTookTime() {
     public void markEndQuery() {
         assert relativeStartNanos != null : "Relative start time must be set when markEndQuery is called";
         overallTook = new TimeValue(System.nanoTime() - relativeStartNanos, TimeUnit.NANOSECONDS);
+        if (isPartial == false) {
+            isPartial = clusterInfo.values().stream().anyMatch(c -> c.failedShards != null && c.failedShards > 0);
+        }
```

Comment:
We seem to handle setting `isPartial` in two places - here and at the end of `innerExecute`. We may want to make it so that it's handled in a single place.

Reply:
++ I combined these two in https://github.com/elastic/elasticsearch/pull/121942/commits/72bc1ea06679d162bb4bb5157b5425ddeaa7243a